### PR TITLE
REMOTE: Added possibility to change target IP of remote API

### PR DESCRIPTION
**Pull request by Specker**
#  Description 
Added possibility to change target of the Remote API to allow development of scripts on other device than where game is running.

It is a niche situation but with Steam Deck gaining popularity and being a good device for idlers, I think giving this possibility for the more savvy would be great.

## Possible usage
- Game running on different device(Steam Deck, laptop, vm)
- File synchronization utility([Viteburner](https://github.com/Tanimodori/viteburner-template),[bitburner-filesync](https://github.com/bitburner-official/bitburner-filesync)) on main device(main pc, work pc)

# Features
- possibility to change target IP of the Remote API

image
image

### CODE REVIEW

**Comment by Hoekstraa** [comment]
Note: This wasn't a feature up until now due to potential security risks. If this gets accepted without a security assessment, I won't take responsibility for anything that happens due to this change.

File: src/GameOptions/ui/RemoteAPIPage.tsx [hunk 1]
```diff
@@ -28,6 +34,25 @@ export const RemoteAPIPage = (): React.ReactElement => {
         </Link>
       </Typography>
       <ConnectionBauble isConnected={isRemoteFileApiConnectionLive} />
+      <Tooltip
+        title={
+          <Typography>
+            This address is used to connect to a Remote API port, please ensure that it matches with your Remote API
```
Comment (d0sboots) [inline src/GameOptions/ui/RemoteAPIPage.tsx]: This was copied from port, should be updated for address.

File: src/RemoteFileAPI/RemoteFileAPI.ts [hunk 2]
```diff
@@ -5,8 +5,9 @@ let server: Remote;
 
 export function newRemoteFileApiConnection(): void {
   if (server) server.stopConnection();
-  if (Settings.RemoteFileApiPort === 0) return;
-  server = new Remote("localhost", Settings.RemoteFileApiPort);
+  if (Settings.RemoteFileApiAddress === "") Settings.RemoteFileApiAddress = "localhost";
```
Comment (d0sboots) [inline src/RemoteFileAPI/RemoteFileAPI.ts]: Don't change the value in settings, that's (potentially) permanent. Set a local instead.

**Review by d0sboots — COMMENTED** [review]
Although it's wise to be cautious, I don't see any security issues here. It's an outbound connection; any script can make those at any time through various built-in JS APIs. And the configuration is player controlled; a sufficiently exploity script could change the address, but it's possible that a sufficiently exploity script could have done that *anyway*.

File: src/RemoteFileAPI/RemoteFileAPI.ts (hunk 2 above)
Comment (Specker) [reply src/RemoteFileAPI/RemoteFileAPI.ts]: I was thinking about this line as more of return to default value in settings and connect. 
Same way the port field defaults to 0 when you empty the field. But string field would return to localhost when you remove 'l'.

Comment (Specker) [reply src/RemoteFileAPI/RemoteFileAPI.ts]: Switched logic to display "localhost" placeholder instead switching it in Settings. 
Connection with empty string to localhost is now done with Ternary.

**Review by d0sboots — COMMENTED** [review]
Looks good, but @Snarling has the final say.

**Comment by Snarling** [comment]
Sorry for the late merge @Specker , I was conflicted when I first saw this. In the end I agree with @d0sboots analysis above.

**Pull request merged**